docs/content/tutorial/step_04.ngdoc

```diff
@@ -1,199 +1,310 @@
+component is defined, we place it inside `app/phone-list/`.
 
-Let's turn our attention to the end-to-end test.
+We copied the contents of the `template` property (the HTML code) into
+`app/phone-list/phone-list.template.html` and modified our CDO like this:
 
-__`test/e2e/scenarios.js`:__
+<br />
+**`app/phone-list/phone-list.component.js`:**
 
 ```js
-...
-    it('should be possible to control phone order via the drop down select box', function() {
+angular.
+  module('phoneList').
+  component('phoneList', {
+    // Note: The URL is relative to our `index.html` file
+    templateUrl: 'phone-list/phone-list.template.html',
+    controller: ...
+  });
+```
+
+At runtime, when Angular needs to create an instance of the `phoneList` component, it will make an
+HTTP request to get the template from `app/phone-list/phone-list.template.html`.
 
-      var phoneNameColumn = element.all(by.repeater('phone in phones').column('phone.name'));
-      var query = element(by.model('query'));
+<div class="alert alert-info">
+  Keeping inline with our convention, we will be using the `.template` suffix for external
+  templates. Another common convention is to just have the `.html` extension
+  (e.g. `phone-list.html`).
+</div>
 
-      function getNames() {
-        return phoneNameColumn.map(function(elm) {
-          return elm.getText();
-        });
-      }
+<div class="alert alert-warning">
+  <p>
+    Using an external template like this, will result in more HTTP requests to the server (one for
+    each external template). Although Angular takes care not to make extraneous requests (e.g.
+    fetching the templates lazily, caching the results, etc), additional requests do have a cost
+    (especially on mobile devices and data-plan connections).
+  </p>
+  <p>
+    Luckily, there are ways to avoid the extra costs (while still keeping your templates external).
+    A detailed discussion of the subject is outside the scope of this tutorial, but you can take a
+    look at the {@link ng.$templateRequest $templateRequest} and
+    {@link ng.$templateCache $templateCache} services for more info on how Angular manages external
+    templates.
+  </p>
+</div>
 
-      query.sendKeys('tablet'); //let's narrow the dataset to make the test assertions shorter
 
-      expect(getNames()).toEqual([
-        "Motorola XOOM\u2122 with Wi-Fi",
-        "MOTOROLA XOOM\u2122"
-      ]);
+## Final Directory/File Layout
 
-      element(by.model('orderProp')).element(by.css('option[value="name"]')).click();
+After all the refactorings that took place, this is how our application looks like from the outside:
 
-      expect(getNames()).toEqual([
-        "MOTOROLA XOOM\u2122",
-        "Motorola XOOM\u2122 with Wi-Fi"
-      ]);
-    });...
+<br />
+**`/`:**
+
+```
+  app/
+    phone-list/
+      phone-list.component.js
+      phone-list.component.spec.js
+      phone-list.module.js
+      phone-list.template.html
+    app.css
+    app.module.js
+    index.html
```

Comment:
It's because there isn't one yet (see https://github.com/angular/angular.js/pull/14416/files/0234f66cdb8703393ab24bffc49a467b48e912e7#r61547667).